app/helpers/host_description_helper.rb

```diff
@@ -136,6 +136,7 @@ def base_host_overview_buttons(host)
       { :button => link_to_if_authorized(_("Audits"), { :controller => 'audits', :action => 'index', :search => "host = #{host.name}" }, :title => _("Host audit entries"), :class => 'btn btn-default'), :priority => 100 },
       ({ :button => link_to_if_authorized(_("Facts"), hash_for_host_facts_path(:host_id => host), :title => _("Browse host facts"), :class => 'btn btn-default'), :priority => 200 } if host.fact_values.any?),
       ({ :button => link_to_if_authorized(_("Reports"), hash_for_host_config_reports_path(:host_id => host), :title => _("Browse host config management reports"), :class => 'btn btn-default'), :priority => 300 } if host.reports.any?),
+      { :button => link_to(_("New UI"), '#', :title => _("Switching to the new host details UI"), :onclick => "tfm.nav.pushUrl('#{host_details_page_path(host.name)}')", :class => 'btn btn-default'), :priority => 50 },
```

Comment:
Why not use the link directly instead of `onClick` handler? That forces a full page reload and also clears the tooltip.
also small change to the tooltip message
```suggestion
      { :button => link_to(_("New UI"), host_details_page_path(host.name), :title => _("Switch to the new host details UI"), :class => 'btn btn-default'), :priority => 50 },
```

Reply:
Mainly because of these two reasons: 
1.  https://github.com/theforeman/foreman/pull/8780 removes the experimental route and adds a redirect mechanism for switching between legacy <--> new UI page. we will need a react-router transition for such behavior 
2.  It's a much faster transition than a full page reload

Though for the current version I agree we can  keep using the full page reload